src/main/cms/cms.c

```diff
@@ -259,17 +275,69 @@ static void cmsPadToSize(char *buf, int size)
 
     buf[size] = 0;
 }
+#endif
+
+// Pad buffer to the left, i.e. align right
+static void cmsPadLeftToSize(char *buf, int size)
+{
+    int i,j;
+    int len = strlen(buf);
+
+    for (i = size-1, j = size-len ; i-j >= 0 ; i--) {
```

Comment:
```
memmove(buf + size - len, buf, len+1);
memset(buf, ' ', size-len);

Reply:
Hmm, no. Does not work with overlaps.

Reply:
memmove is guaranteed to handle overlaping memory areas